docs/adrs/0096-hook-extensions.md

```diff
@@ -0,0 +1,32 @@
+# ADR 0096: Hook extensions
+
+**Date:** 3 August 2023
+
+**Status**: Proposed <!--Accepted|Rejected|Superceded|Deprecated-->
+
+## Context
+
+The current implementation of container hooks does not allow users to customize the pods created by the hook. While the implementation is designed to be used as is or as a starting point, building and maintaining a custom hook implementation just to specify additional fields is not a good user experience.
+
+## Decision
+
+We have decided to add hook extensions to the container hook implementation. This will allow users to customize the pods created by the hook by specifying additional fields. The hook extensions will be implemented in a way that is backwards-compatible with the existing hook implementation.
+
+To allow customization, the runner executing the hook should have `ACTIONS_RUNNER_CONTAINER_HOOK_JOB_TEMPLATE` environment variable pointing to a yaml file on the runner system.
```

Comment:
Should we specify a different template for container steps? @fhammerl @thboop 
I'm not sure, but maybe they have different security requirements

Reply:
We spoke in person, lets stick with 1 template for both and we can add individual templates in the future as needed :)

Reply:
That may mean we want to rename this variable to `ACTIONS_RUNNER_CONTAINER_HOOK_TEMPLATE`